ai-security-guard 流式处理bugfix || ai-security-guard streaming bugfix

Thomas-Eliot opened the pull request:
<!-- Please make sure you have read and understood the contributing guidelines -->

## Ⅰ. Describe what this PR did
- config.go 新增/扩展缓冲区刷新时间间隔配置项，支持对流式输出刷新频率进行控制
- openai.go 重构 OpenAI 文本流式响应的读取与写入逻辑，引入基于时间间隔的缓冲刷新
- 改善长文本、慢速流式场景下的响应体验，减少卡顿和不完整输出的情况

## Ⅱ. Does this pull request fix one issue?
<!-- If that, add "fixes #xxx" below in the next line, for example, fixes #97. -->
fixes #3158 

## Ⅲ. Why don't you add test cases (unit test/integration test)? 


## Ⅳ. Describe how to verify it
本地编译验证流程

## Ⅴ. Special notes for reviews

<!--This is a translation content dividing line, the content below is generated by machine, please do not modify the content below-->
---
<!-- Please make sure you have read and understood the contributing guidelines -->

## Ⅰ. Describe what this PR did
- config.go adds/expands the buffer refresh interval configuration item to support controlling the streaming output refresh frequency
- openai.go reconstructs the reading and writing logic of OpenAI text streaming response and introduces buffer refresh based on time interval
- Improve the response experience in long text and slow streaming scenarios, reducing lags and incomplete output

## Ⅱ. Does this pull request fix one issue?
<!-- If that, add "fixes #xxx" below in the next line, for example, fixes #97. -->
fixes #3158

## Ⅲ. Why don't you add test cases (unit test/integration test)?


## Ⅳ. Describe how to verify it
Local compilation verification process

## Ⅴ. Special notes for reviews

rinfx commented:
为什么会通过设置刷新频率来控制流式输出，现在的逻辑是只要chunk已经检查完了就会返回给客户端

rinfx commented on plugins/wasm-go/extensions/ai-security-guard/config/config.go:
```diff
@@ -361,7 +365,8 @@ func (config *AISecurityConfig) SetDefaultValues() {
 	config.Timeout = DefaultTimeout
-	config.BufferLimit = 1000
+	config.BufferLimit = 200
+	config.BufferFlushTimeInterval = 1000
```
flushInterval的默认值调大一点吧，不然对于现有用户来说，新插件发布很容易出现行为不符合预期的情况

Thomas-Eliot committed:
fix: ai-security-guard 流式处理bugfix

- 添加缓冲区刷新时间间隔(bufferFlushTimeInterval)配置支持
- 优化流式响应处理，支持按时间窗口和缓冲区大小分批触发安全检查
- 修改 BufferLimit 默认值为 200，BufferFlushTimeInterval 默认不启用
- 增加流式响应分批触发安全检查测试用例

Signed-off-by: Thomas-Eliot <lvshui.wqs@alibaba-inc.com>

Thomas-Eliot committed:
Merge branch 'main' into hotfix/issue-3158-new
